library/core/src/fmt/num.rs

```diff
@@ -330,10 +330,17 @@ macro_rules! impl_Exp {
                 if subtracted_precision != 0 {
                     let rem = n % 10;
                     n /= 10;
+                    let last_digit_odd = (n % 10) % 2 != 0;
```

Comment:
You don't need to isolate the last digit to know that it's odd.
```suggestion
                    let last_digit_odd = n % 2 != 0;
```